pkg/chunked/storage_linux.go

```diff
@@ -1141,6 +1141,32 @@ func makeEntriesFlat(mergedEntries []fileMetadata) ([]fileMetadata, error) {
 	return new, nil
 }
 
+// checkNoStreams checks that there are no other streams on the channel and that there are no errors.
+func checkNoStreams(streams chan io.ReadCloser, errs chan error) error {
+	var err error
+	closed := 0
+	for {
+		select {
+		case _, ok := <-streams:
+			if !ok {
+				closed++
+			} else if err == nil {
```

Comment:
Nothing guarantees that we read the error from `errs` before we read from this channel. `select` is explicitly designed to choose randomly.